packages/libs/error-stack/src/fmt/color.rs

```diff
@@ -0,0 +1,153 @@
+    /// # #![cfg_attr(not(rust_1_65), allow(dead_code, unused_variables, unused_imports))]
+    /// use std::io::{Error, ErrorKind};
+    /// use owo_colors::OwoColorize;
+    ///
+    /// use error_stack::{report, Report};
+    /// use error_stack::fmt::ColorMode;
+    ///
+    /// struct Suggestion(&'static str);
+    ///
+    /// Report::install_debug_hook::<Suggestion>(|Suggestion(value), context| {
+    ///     let body = format!("suggestion: {value}");
+    ///     match context.color_mode() {
+    ///         ColorMode::Color => context.push_body(body.green().to_string()),
+    ///         ColorMode::Emphasis => context.push_body(body.italic().to_string()),
+    ///         ColorMode::None => context.push_body(body)
+    ///     };
+    /// });
+    ///
+    /// let report =
+    ///     report!(Error::from(ErrorKind::InvalidInput)).attach(Suggestion("oh no, try again"));
+    ///
+    /// # fn render(value: String) -> String {
+    /// #     let backtrace = regex::Regex::new(r"backtrace no\. (\d+)\n(?:  .*\n)*  .*").unwrap();
+    /// #     let backtrace_info = regex::Regex::new(r"backtrace( with (\d+) frames)? \((\d+)\)").unwrap();
+    /// #
+    /// #     let value = backtrace.replace_all(&value, "backtrace no. $1\n  [redacted]");
+    /// #     let value = backtrace_info.replace_all(value.as_ref(), "backtrace ($3)");
+    /// #
+    /// #     ansi_to_html::convert_escaped(value.as_ref()).unwrap()
+    /// # }
+    /// #
+    /// Report::set_color_mode(Some(ColorMode::None));
+    /// println!("{report:?}");
+    /// # #[cfg(rust_1_65)]
+    /// # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__preference_none.snap")].assert_eq(&render(format!("{report:?}")));
+    ///
+    /// Report::set_color_mode(Some(ColorMode::Emphasis));
+    /// println!("{report:?}");
+    /// # #[cfg(rust_1_65)]
+    /// # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__preference_emphasis.snap")].assert_eq(&render(format!("{report:?}")));
+    ///
+    /// Report::set_color_mode(Some(ColorMode::Color));
+    /// println!("{report:?}");
+    /// # #[cfg(rust_1_65)]
+    /// # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__preference_color.snap")].assert_eq(&render(format!("{report:?}")));
+    /// ```
+    ///
+    /// Which will result in something like:
+    ///
+    /// <pre>
+    #[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__preference_none.snap"))]
+    /// </pre>
+    ///
+    /// <pre>
+    #[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__preference_emphasis.snap"))]
+    /// </pre>
+    ///
+    /// <pre>
+    #[doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__preference_color.snap"))]
+    /// </pre>
+    #[cfg(feature = "color")]
```

Comment:
Why this is gated behind a feature flag? I expect that downstream binaries want to set `ColorMode::None` or `ColorMode::Emphasis` and downstream libraries can pick up those values, this has nothing to do on our dependency on `owo-colors`.

Reply:
fixed in [f83228f](https://github.com/hashintel/hash/pull/1800/commits/f83228f7d0c91828e09c642bd916119794d17402)